### ISSUE TITLE
Problems uploading 1.5 mb file

### ISSUE THREAD

**Issue by g-gabber** [scottgonzalez/node-wordpress#37]
I'm tring to upload a gif file of 1.5 mb size,
the upload call blocks for 1 minute and then returns

{ '0': [Error: Unknown XML-RPC tag 'TITLE'], '1': undefined }.

I tried to wireshark the connection. The request (see below, i removed base64) seems to be ok, but there is no response from server.
```xml
<?xml version="1.0"?>
<methodCall>
	<methodName>wp.uploadFile</methodName>
	<params>
		<param>
			<value>
				<int>0</int>
			</value>
		</param>
		<param>
			<value>
				<string>xxxx</string>
			</value>
		</param>
		<param>
			<value>
				<string>xxxxx</string>
			</value>
		</param>
		<param>
			<value>
				<struct>
					<member>
						<name>name</name>
						<value>
							<string>xxxx.gif</string>
						</value>
					</member>
					<member>
						<name>type</name>
						<value>
							<string>image/gif</string>
						</value>
					</member>
					<member>
						<name>bits</name>
						<value>
							<base64></base64>
						</value>
					</member>
					<member>
						<name>overwrite</name>
						<value>
							<boolean>1</boolean>
						</value>
					</member>
					<member>
						<name>post_id</name>
						<value/>
					</member>
				</struct>
			</value>
		</param>
	</params>
</methodCall>
```

**Comment by g-gabber** [comment]
seems to be server problem, because I get following response

<html>
<head><title>504 Gateway Time-out</title></head>
<body bgcolor="white">
<center><h1>504 Gateway Time-out</h1></center>
<hr><center>nginx</center>
</body>
</html>

**Comment by g-gabber** [comment]
ok I found the problem, my provider use ModSecurity. And request body is limited to not larger than the configured limit (1048576).

**Comment by g-gabber** [comment]
I disabled ModSecurity and every thing is now working

**Comment by scottgonzalez** [comment]
Can you do me a favor? Add the following block of code inside https://github.com/scottgonzalez/node-wordpress/blob/619852a563ece0fc44bda35477ae56f4d49d93ec/lib/wordpress.js#L82

```js
console.log(error);
for (var prop in error) {
  console.log(prop, error[prop]);
}
```

Re-enable ModSecurity, retry that request and let me know what gets logged.

Hopefully there will be some information that I can use to report a better error.

**Comment by g-gabber** [comment]
yes sure

**Comment by g-gabber** [comment]
![image](https://cloud.githubusercontent.com/assets/4931017/15775010/bb74fc88-297f-11e6-99ff-bc89ebedbc5e.png)
I set breakpoint at this callback, and in my view you can forward statusCode and statusMessage from the response and maybe body as well.

**Comment by scottgonzalez** [comment]
Thanks! Given the structure of that error object, I'm not sure that I can really provide too much more value. All of the information anybody would ever need is already there and if I create a new event object and only pass through some of the properties, then the end user is losing potentially useful information.

What do you think about just prepending the status message to the main message? So in this case, you would have gotten the following output:

`{ '0': [Error: (Gateway Time-out) Unknown XML-RPC tag 'TITLE'], '1': undefined }.`

**Comment by scottgonzalez** [comment]
Since most people won't realize there are additional properties on the error object, we could also make this more verbose, something like:

`Unknown XML-RPC tag 'TITLE' (caused by: Gateway Time-out; see error.body and error.res for more information`

**Comment by g-gabber** [comment]
Yeah, it is a good question how to design the api.
I think it would be better just to extend the error object with statusCode.

**Comment by scottgonzalez** [comment]
> I think it would be better just to extend the error object with statusCode.

The error object already has `error.res.statusCode`. Did you mean just to prepend the `statusMessage` as shown in https://github.com/scottgonzalez/node-wordpress/issues/37#issuecomment-223579644?

**Comment by g-gabber** [comment]
yeah, it is additional hint for user why not.

**Comment by scottgonzalez** [comment]
Updated error messages are available in v1.2.1. Thanks for helping dig into this.
